prover/crates/bin/prover_job_monitor/src/task_wiring.rs

```diff
@@ -0,0 +1,86 @@
+use std::time::Duration;
+
+use anyhow::Context;
+use tracing::Instrument;
+use zksync_prover_dal::{Connection, ConnectionPool, Prover};
+
+/// Task trait to be run in ProverJobMonitor.
+#[async_trait::async_trait]
+pub trait Task {
+    async fn invoke(&self, connection: &mut Connection<Prover>) -> anyhow::Result<()>;
+}
+
+/// Wrapper for Task with a periodic interface. Holds information about the task and provides DB connectivity.
+struct PeriodicTask {
+    job: Box<dyn Task + Send + Sync>,
+    name: String,
+    interval: Duration,
+}
+
+impl PeriodicTask {
+    async fn run(
+        &self,
+        mut stop_receiver: tokio::sync::watch::Receiver<bool>,
+        connection_pool: ConnectionPool<Prover>,
+    ) -> anyhow::Result<()> {
+        tracing::info!(
+            "Started Task {} with run interval: {:?}",
+            self.name,
+            self.interval
+        );
+        while !*stop_receiver.borrow_and_update() {
+            let mut connection = connection_pool
+                .connection()
+                .await
+                .context("failed to get database connection")?;
+            self.job
+                .invoke(&mut connection)
+                .instrument(tracing::info_span!("run", service_name = %self.name))
+                .await
+                .context("failed to invoke task")?;
+            // Error here corresponds to a timeout w/o `stop_receiver` changed; we're OK with this.
+            tokio::time::timeout(self.interval, stop_receiver.changed())
+                .await
+                .ok();
```

Comment:
a nit: you probably would want a better expression of a periodic task with a ready to use primitive [tokio::time::interval](https://docs.rs/tokio/latest/tokio/time/fn.interval.html) + couple it with select over it and stop condition. That would be a dash more idiomatic and tell the reader faster whats going on